src/lib/plugins/chest.js

```diff
@@ -1,26 +1,103 @@
+      // TODO: Large chest (NOT IMPLEMENTED)
+      // Shulker boxes
+      const blockShulkerBox = mcData.blocksByName.shulker_box
+      const blockShulkerBoxRed = mcData.blocksByName.red_shulker_box
+      const blockShulkerBoxOrange = mcData.blocksByName.orange_shulker_box
+      const blockShulkerBoxYellow = mcData.blocksByName.yellow_shulker_box
+      const blockShulkerBoxLime = mcData.blocksByName.lime_shulker_box
+      const blockShulkerBoxGreen = mcData.blocksByName.green_shulker_box
+      const blockShulkerBoxLightBlue = mcData.blocksByName.light_blue_shulker_box
+      const blockShulkerBoxCyan = mcData.blocksByName.cyan_shulker_box
+      const blockShulkerBoxBlue = mcData.blocksByName.blue_shulker_box
+      const blockShulkerBoxPurple = mcData.blocksByName.purple_shulker_box
+      const blockShulkerBoxMagenta = mcData.blocksByName.magenta_shulker_box
+      const blockShulkerBoxBrown = mcData.blocksByName.brown_shulker_box
+      const blockShulkerBoxGray = mcData.blocksByName.gray_shulker_box
+      const blockShulkerBoxLightGray = mcData.blocksByName.light_gray_shulker_box
+      const blockShulkerBoxBlack = mcData.blocksByName.black_shulker_box
+      const blockShulkerBoxWhite = mcData.blocksByName.white_shulker_box
+      const blockShulkerBoxPink = mcData.blocksByName.pink_shulker_box
+      const blockShulkerBoxes = [blockShulkerBox.id, blockShulkerBoxRed.id, blockShulkerBoxOrange.id, blockShulkerBoxYellow.id,
+        blockShulkerBoxLime.id, blockShulkerBoxGreen.id, blockShulkerBoxLightBlue.id, blockShulkerBoxCyan.id,
+        blockShulkerBoxBlue.id, blockShulkerBoxPurple.id, blockShulkerBoxMagenta.id, blockShulkerBoxBrown.id,
+        blockShulkerBoxGray.id, blockShulkerBoxLightGray.id, blockShulkerBoxBlack.id, blockShulkerBoxWhite.id,
+        blockShulkerBoxPink.id]
+      // Playing sound of opening chest
+      if (blockAbove) { return }
+      serv.playSound('block.chest.open', player.world, block.position, {})
+      // Dynamic window ID feature
+      if (player.windowId === undefined) { player.windowId = 1 } else { player.windowId = player.windowId + 1 }
+      let chestWindowTitle = 'Unknown'
+      if (id === blockChest.id) {
+        chestWindowTitle = { translate: 'container.chest' }
+        player.windowType = 'chest'
       }
+      if (id === blockEnderChest.id) {
+        chestWindowTitle = { translate: 'container.enderchest' }
+        player.windowType = 'ender_chest'
+      }
+      // TODO: Large chest (NOT IMPLEMENTED)
+      if (blockShulkerBoxes.includes(id)) {
+        chestWindowTitle = { translate: 'container.shulkerBox' }
+        player.windowType = 'shulker_box'
+      }
+      player.windowPos = block.position
+      // Opening chest GUI window
+      player._client.write('open_window', {
+        windowId: player.windowId,
+        inventoryType: 2,
+        windowTitle: JSON.stringify(chestWindowTitle)
+      })
+      // Sending chest content (NOT IMPLEMENTED)
```

Comment:
Looks like sending the chunk content is not implemented

Reply:
Yes, it isn't. I can't implement it now because if the content isn't actually stored then I have nothing to send

Reply:
it can be stored in the world in memory, it doesn't have to be stored on the disk